app/src/services/bucket.js

```diff
@@ -40,19 +91,16 @@ const service = {
         createdBy: data.userId
       };
 
-      // TODO: Add conflict/merge logic to avoid bucket duplicates?
-
       const response = await Bucket.query(trx).insert(obj).returning('*');
 
-      // TODO: Add permissions when all fields match?
       // Add all permission codes for the uploader
-      // if (data.userId) {
-      //   const perms = Bucket.values(Permissions).map((p) => ({
-      //     userId: data.userId,
-      //     permCode: p
-      //   }));
-      //   await permissionService.addPermissions(data.id, perms, data.userId, trx);
-      // }
+      if (data.userId) {
+        const perms = Object.values(Permissions).map((p) => ({
+          userId: data.userId,
+          permCode: p
+        }));
+        await bucketPermissionService.addPermissions(data.bucketId, perms, data.userId, trx);
```

Comment:
Think the bucketID passed to here is undefined (taking from `data` when it should be `obj` or `response`). At least that's what my debugger is doing

Reply:
Good catch! It should be `obj.bucketId`.